Sources/TuistKit/Utils/XCResultParser.swift

```diff
@@ -0,0 +1,69 @@
+import Foundation
+import Mockable
+import Path
+import XCResultKit
+
+@Mockable
+public protocol XCResultParsing {
+    func parse(path: AbsolutePath) -> ParsedXCResult?
+}
+
+public struct ParsedXCResult {
+    let passingTestTargetNames: Set<String>
```

Comment:
I'd change the properties of the `XCResult` to more closely follow the models of the underlying library.

Something like:
```suggestion
    let invocationRecord: InvocationRecord?
    
    struct InvocationRecord {
      struct Action {...}
      let actions: [Action]
    }
```

The `successfulTestTargets` can then be an extension of `XCResult` or in the `TestService`. In other words, I don't think `XCResultParser` should "interpret" the results, it should be a simple parser that returns a model that more-or-less represents the underlying data structure.